packages/sdk/src/utils.ts

```diff
@@ -5,21 +5,64 @@ export const omit = <TR>(ks: string[], original: object) =>
 
 type RetryCallback = () => Promise<any>;
 
-export const retryTimes = (n: number, fn: RetryCallback) =>
-  new Promise((resolve, reject) =>
-    retry(n, fn)
-      .then((result) => resolve(result))
-      .catch((err) => reject(err))
-  );
-
-const retry = (i: number, fn: RetryCallback) =>
-  fn().catch((err) => {
-    if (!err) throw err;
-    if (i <= 0) throw err;
-    return retry(i - 1, fn);
-  });
-
 export const generateRID = (() => {
   let rid = 1;
   return () => rid++ && String(rid);
 })();
+
+const failureTreshold = 3
+const retryTime = 180000 // 3 minutes;
+
+const updateCircuitBreakerStatus = (failureCount: number, key: string) => {
+  sessionStorage.setItem(key, JSON.stringify({ failureCount, lastFailureDate: new Date() }))
+}
+
+const doRequest = (attempt: number = 0, fn: RetryCallback, key: string) => {
+  return fn().catch((err) => {
+    const failureCount = attempt + 1
+    updateCircuitBreakerStatus(failureCount, key);
+    if (err.response.status < 500 || failureCount >= failureTreshold) throw err;
+    return doRequest(failureCount, fn, key);
+  })
+}
+
+/**
+ * 
+ * @param key widgetId + MerchantId - key of the session storage value
+ * @returns failureCount and lastFailureDate of the matched key
+ */
+const getCircuitBreakerStatus = (key: string): { failureCount: number; lastFailureDate: Date; } => {
+  const findifyCircuitBreaker = sessionStorage.getItem(key);
+  return findifyCircuitBreaker ? JSON.parse(findifyCircuitBreaker) : { failureCount: null, lastFailureDate: null };
+}
+
+const tryAgainCheck = (lastFailure) => {
+  const lastAttempt = new Date(lastFailure);
```

Comment:
Handle error if variable is not date, this would throw an error

Reply:
Just added a lastAttempty?.getTime() so if it is invalid at least there is no error.
I do not expect this to not be a date